src/jit/gentree.cpp

```diff
@@ -17982,6 +17982,28 @@ bool GenTree::isCommutativeSIMDIntrinsic()
 #endif // FEATURE_SIMD
 
 #ifdef FEATURE_HW_INTRINSICS
+bool GenTree::isCommutativeHWIntrinsic() const
+{
+#ifdef _TARGET_XARCH_
+    assert(gtOper == GT_HWIntrinsic);
+    HWIntrinsicFlag flags = Compiler::flagsOfHWIntrinsic(AsHWIntrinsic()->gtHWIntrinsicId);
+    return ((flags & HW_Flag_Commutative) != 0);
+#else
+    return false;
```

Comment:
FYI. @sdmaclea. This will probably need to be handled for ARM eventually.

Reply:
We should assert `numArgs == 2` if `(flags & HW_Flag_Commutative) != 0`.

Reply:
Will do.

Reply:
@fiigii, actually can't assert that here, since the method needs to be `const` to work upstream.